libra-swarm/src/swarm.rs

```diff
@@ -286,32 +283,53 @@ impl LibraSwarm {
                     .expect("unable to create temporary config dir");
                 LibraSwarmDir::Temporary(temp_dir)
             }
-        };
-        println!("Base directory containing logs and configs: {:?}", &dir);
-        dir
+        }
     }
 
     pub fn configure_swarm(
         num_nodes: usize,
         role: RoleType,
-        faucet_account_keypair: KeyPair<Ed25519PrivateKey, Ed25519PublicKey>,
         config_dir: Option<String>,
         template_path: Option<String>,
         upstream_config_dir: Option<String>,
     ) -> Result<LibraSwarm> {
         let swarm_config_dir = Self::setup_config_dir(&config_dir);
-        let mut config_builder = SwarmConfigBuilder::new();
-        if let Some(template_path) = template_path {
-            config_builder.with_base(utils::workspace_root().join(template_path));
-        }
-        config_builder
-            .with_ipv4()
-            .with_num_nodes(num_nodes)
-            .with_output_dir(&swarm_config_dir)
-            .with_faucet_keypair(faucet_account_keypair)
-            .with_role(role)
-            .with_upstream_config_dir(upstream_config_dir.clone());
-        let config = config_builder.build()?;
+        let mut template = if let Some(template_path) = template_path {
+            NodeConfig::load(utils::workspace_root().join(template_path)).unwrap_or_default()
+        } else {
+            let mut template = NodeConfig::default();
+            template.vm_config.publishing_options = VMPublishingOption::Open;
+            template
+        };
+        template.base.role = role;
+
+        let config_path = &swarm_config_dir.as_ref().to_path_buf();
+        let config = if role.is_validator() {
+            let mut validator_builder = ValidatorConfig::new();
+            validator_builder.template(template).nodes(num_nodes);
+            SwarmConfig::build(&validator_builder, config_path)?
+        } else {
+            let upstream_config_dir_str =
```

Comment:
same thing here, re-defining `upstream_config_dir` instead of declaring a new variable would be much clearer.